### PR TITLE
Migrate to flow v0.54.0

### PR DESCRIPTION
**Pull request by Hypnosphi**
Flow has recently introduced a lot of improvements in field of React: https://flow.org/en/docs/react/

### CODE REVIEW

File: src/prop-converter.js [hunk 1]
```diff
@@ -72,27 +75,43 @@ function propConverter(
...
     // eslint-disable-next-line class-methods-use-this
-    checkChildren(children: mixed) {
+    checkChildren(children) {
       // Skip all console warnings in production.
       // Bail early, to avoid unnecessary work.
       if (isProduction()) {
         return;
       }
 
+      // same as React.Node, but without fragments, see https://github.com/facebook/flow/issues/4781
```
Comment (tobilen) [inline src/prop-converter.js]: very nice catch

File: flow-typed/react-flip-move_v2.9.x.js [hunk 2]
```diff
@@ -77,11 +70,16 @@ declare module 'react-flip-move' {
...
+    staggerDurationBy: string | number,
+    staggerDelayBy: string | number,
+    enterAnimation: AnimationProp,
+    leaveAnimation: AnimationProp,
     appearAnimation?: AnimationProp,
     disableAnimations?: boolean, // deprecated, use disableAllAnimations instead
   };
 
-  declare export default Class<React$Component<FlipMoveDefaultProps, FlipMoveProps, void>>;
+  declare export default ComponentType<FlipMoveProps>;
```
Comment (tobilen) [inline flow-typed/react-flip-move_v2.9.x.js]: 😍

Comment (Hypnosphi) [reply flow-typed/react-flip-move_v2.9.x.js]: Seems like we have to revert this part for now =( https://github.com/flowtype/flow-typed/issues/1209

File: test/index.spec.js [hunk 3]
```diff
@@ -331,9 +331,9 @@ Please wrap your components in a native element (eg. <div>), or a non-functional
         expect(warnStub).to.have.been.calledWith(`
 >> Error, via react-flip-move <<
 
-You provided a plain text node as a child to <FlipMove>. Flip Move needs containers with unique keys to move children around.
+You provided a primitive (text, number, or boolean) node as a child to <FlipMove>. Flip Move needs containers with unique keys to move children around.
```
Comment (joshwcomeau) [inline test/index.spec.js]: Awesome :+1:

File: src/error-messages.js [hunk 4]
```diff
@@ -19,12 +19,12 @@ You provided a stateless functional component as a child to <FlipMove>. Unfortun
...
 
-export const textNodeSupplied = warnOnce(`
+export const primitiveNodeSupplied = warnOnce(`
 >> Error, via react-flip-move <<
 
-You provided a plain text node as a child to <FlipMove>. Flip Move needs containers with unique keys to move children around.
+You provided a primitive (text, number, or boolean) node as a child to <FlipMove>. Flip Move needs containers with unique keys to move children around.
 
-Please wrap your text in a native element (eg. <span>), or a component.
+Please wrap your value in a native element (eg. <span>), or a component.
```
Comment (joshwcomeau) [inline src/error-messages.js]: Awesome :+1:

**Comment by joshwcomeau** [comment]
Looks great! Digging the modern Flow improvements. Thanks @Hypnosphi, and thanks @tobilen for the review :)

**Pull request merged**